Fixing updating of TPC calibs from CCDB

matthias-kleiner opened the pull request:
- moving TPC code to separate functions

matthias-kleiner committed:
Fixing updating of TPC calibrations

fixing allocation of memory for topology correction in case residual calib is set from file

wiechula commented:
@davidrohr , wouldn't it in general be nicer to convert the workflow to a class inheriting from `Task`? This might look more structured.

davidrohr commented on GPU/Workflow/src/GPUWorkflowSpec.cxx:
```diff
@@ -89,6 +89,45 @@ using namespace o2::tpc;
+void fetchCalibsCCDBTPC(ProcessAttributes* processAttributes, gpuworkflow::Config const& specconfig, ProcessingContext& pc);
+
+/// storing the new calib objects by overwritting the old calibs
+void setCalibsTPC(ProcessAttributes* processAttributes);
```
perhaps we can find a better name here, like `storeUpdatedCalibsTPCPtrs` instead of `setCalibsTPC` to make it clear what happens here.

davidrohr reviewed (changes requested):
This is exactly what I meant in the mail exchange, except for one minor comment on a misleading function name below.

matthias-kleiner commented:
> This is exactly what I meant in the mail exchange, except for one minor comment on a misleading function name below.

Okay, thanks for checking the code. I renamed the function and squashed the commits

davidrohr commented:
> @davidrohr , wouldn't it in general be nicer to convert the workflow to a class inheriting from `Task`? This might look more structured.

Yes, I agree, but I'd try to get this in first, then we can try to change it perhaps. But migrating to a task was not a priority yet, since I don't see any benefit except for being more in line with other workflows. Note this one was developed before this task class was available.
